homeassistant/components/nmbs/sensor.py

```diff
@@ -104,7 +108,12 @@ def name(self):
     @property
     def unique_id(self):
         """Return a unique ID."""
-        return self._unique_id
+        unique_id = f"{self._station}_{self._station_from}_{self._station_to}"
+
+        if self._name:
+            unique_id = f"{self._station}_{self.name}"
```

Comment:
Name is user configurable and not allowed to be part of unique_id.

https://developers.home-assistant.io/docs/en/entity_registry_index.html#unique-id-requirements